src/main/java/Storage.java

```diff
@@ -0,0 +1,171 @@
+        }
+
+        writeToFileWrapper(formattedTask);
+    }
+
+
+    /**
+     * Hides the try catch block whilst executing the method
+     * writeToFile.
+     * @param textToAdd The text to be added to the task list.
+     */
+    private void writeToFileWrapper(String textToAdd) {
+        try {
+            writeToFile(filePath, textToAdd);
+        } catch (IOException e) {
+            System.out.println(e);
+        }
+    }
+
+    /**
+     * Writes given input to given file path.
+     * @param filePath The path to the text file to be written to.
+     * @param textToAdd The text to be added to the text file.
+     * @throws IOException The IO Exception.
+     */
+    private void writeToFile(String filePath, String textToAdd) throws IOException {
+        FileWriter fileWriter = new FileWriter(filePath);
+        fileWriter.write(textToAdd + "\n");
+        fileWriter.close();
+    }
+
+    /**
+     * Hides the try catch block whilst executing the method
+     * transferContentsToList.
+     */
+    private void transferContentsToListWrapper(TaskList taskList) {
+        try {
+            transferContentsToList(filePath, taskList);
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found");
+        }
+    }
+
+    /**
+     * Transfers the contents of the file to a static array list.
+     * @param filePath The path to the source text file.
+     * @throws FileNotFoundException The exception to be thrown if no such file is found.
+     */
+    private void transferContentsToList(String filePath, TaskList taskList)
+            throws FileNotFoundException {
+        File file = new File(filePath);
+        Scanner fileReader = new Scanner(file);
+        while (fileReader.hasNext()) {
+            processNextLine(fileReader.nextLine(), taskList);
+        }
+    }
+
+    /**
+     * Processes the next line into proper format.
+     * @param nextLine The line to be processed.
+     */
+    private void processNextLine(String nextLine, TaskList taskList) {
+        String[] splitString = nextLine.split(" ");
+
+        String taskType = splitString[0];
+        String doneStatus = splitString[2];
+
+        if(taskType.equals("T")) {
+
+            String taskDescription = splitString[4];
+
+            for(int i = 5; i < splitString.length; i++) {
+                taskDescription += " " + splitString[i];
+            }
+
+            ToDo toDo = new ToDo(taskDescription);
+
+            if(doneStatus.equals("1")) {
+                toDo.setDone();
+            }
+
+            taskList.addTask(toDo);
+
+        } else {
+
+            int counter = 5;
+            String taskDescription = splitString[4];
+
+            for(int i = 5; i < splitString.length && !splitString[i].equals("|"); i++) {
+                taskDescription += " " + splitString[i];
+                counter++;
+            }
+
+            String dateTime = splitString[++counter];
```

Comment:
Just a personal suggestion, maybe it's better to not use increment operators within the array index for better readability. But again this is just style preference. (see https://stackoverflow.com/questions/971312/why-avoid-increment-and-decrement-operators-in-javascript)